### ISSUE TITLE
Request:  seafile-applet started by Network Manager should show up in the application indicator toolbar.

### ISSUE THREAD

**Issue by deragon** [haiwen/seafile-client#289]
When seafile-applet is started in the background under a Network Manager script, its application indicator does not show up in the application indicator toolbar.

Since I use Seafile for large libraries, I do not want it to start unless my laptop is on particular networks, including my home network.

I have created an elaborate script for Network Manager to detect when to start seafile-applet or not.  However, for the purpose of this issue, here is a small snippet of code to create one for testing purposes: 
# Run this as root

cat <<EOM >/etc/NetworkManager/dispatcher.d/seafile-applet-test
# !/bin/bash

INTERFACE=$1
IFSTATE=$2

if [ "${IFSTATE}" == "up" ]; then
  su user -c seafile-applet &
else
  killall seafile-applet
fi
EOM

chmod 755 /etc/NetworkManager/dispatcher.d/seafile-applet-test

Now under Ubuntu 12.04 LTS and XFCE, that worked marvelously.  When connected to my home network, the application indicator of seafile-applet would appear.

Now I passed to Ubuntu 14.04 LTS with Unity 7 and the application indicator does not appear anymore.  seafile-applet is though running in the background and is transferring files.  I simply have no access to the GUI.

I realize that my request is not usual and few users might use such a sophisticated scenario, but if the fix is easy, it would be nice that it be implemented.

Seafile client 3.1.8
Ubuntu 14.04 LTS with Unity 7

**Comment by dakira** [comment]
I'm pretty sure you need to pass the correct DISPLAY for indicator to show up.

**Comment by deragon** [comment]
No, it is not that.  I already set DISPLAY.  The code I show in this ticket is not the actual code, but a simplified version to show my point.

Here is what I actually execute:

    export DISPLAY=":0"
    su hans -c xhost +local:root
    su hans -c seafile-applet &
    su hans -c gvim &
    echo "home-up() completed."

Usually the call to gvim is not there, but I added for testing purposes.  gvim did show up on my display, but not the seafile-applet indicator.

And as mentionned previously, it worked before under Ubuntu 12.04 and XFCE.  Now it does not work under Ubuntu 14.04 and Unity 7.  I cannot test under XFCE anymore since I moved to Unity.
